terraform-modules/aws/msk/main.tf

```diff
@@ -0,0 +1,164 @@
+locals{
+  years_valid = 10 
+}
+
+resource "aws_cloudwatch_log_group" "msk_cloudwatch_log_group" {
+  name = var.cloudwatch_logs_log_group
+  tags = var.tags
+}
+
+#######################################
+# S3 bucket
+#######################################
+resource "aws_kms_key" "this" {
+  description             = "This key is used to encrypt bucket objects"
+  deletion_window_in_days = 10
+}
+
+resource "aws_s3_bucket" "this" {
+  count  = var.s3_bucket_create ? 1 : 0
+  bucket = var.s3_logs_bucket
+  tags   = var.tags 
+}
```

Comment:
Is this an encrypted bucket?

Reply:
no, 
let's say that if you want that this module can create its own bucket , it is going to be a bucket lite, you have in this way before!!
in terms of data platform bucket , it has encrypted.

Reply:
All S3 bucket has to have encryption enabled that we are creating

Either use the other module for bucket creation in here

Or make this module accept a var input for the bucket and use another bucket you created somewhere else

Reply:
yes, this is the objective , I am receiving the bucket name already created
s3_bucket_create
I added this functionality in your delete code!!

Reply:
The code that you see where it is not encrypted is not where my process goes through. That is code that you had before, I enabled this terraform to send it my own bucket that I was creating in a previous step.
It’s just what I did
you agree with me?
in short : we don’t have an unencrypted bucket in dp.